grype/db/v6/vulnerability_provider.go

```diff
@@ -35,6 +35,7 @@ var _ interface {
 	vulnerability.Provider
 } = (*vulnerabilityProvider)(nil)
 
+// Deprecated: vulnerability.Vulnerability objects now have metadata included
 func (s vulnerabilityProvider) VulnerabilityMetadata(ref vulnerability.Reference) (*vulnerability.Metadata, error) {
```

Comment:
We haven't had a Grype release with the `VulnerabilityProvider` yet -- my vote is just take the `VulnerabilityMetadata` function out now, is there a reason not to?

I am 💯 on board with vulnerability metadata being set on the vulnerability objects when returned from `FindVulnerabilities`, I hadn't made the change to limit the size of my PRs 😄

Reply:
I didn't attempt to solve the case where we're looking back up Metadata objects when there is missing Reference.Internal fields or for RelatedVulnerabilities (which are References). I think it would be another medium lift to finish that off -- but I entirely agree with removing this function/interface, just didn't want to boil the ocean with this PR.